app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingLinksFragment.kt

```diff
@@ -0,0 +1,233 @@
+                when (it) {
+                    LoadState.Initial -> {
+                        viewModel.accept(InfallibleUiAction.Reload)
+                    }
+                    LoadState.Loading -> {
+                        if (!binding.swipeRefreshLayout.isRefreshing) {
+                            binding.progressBar.show()
+                        } else {
+                            binding.progressBar.hide()
+                        }
+                    }
+                    is LoadState.Success -> {
+                        adapter.submitList(it.data)
+                        binding.progressBar.hide()
+                        binding.swipeRefreshLayout.isRefreshing = false
+                        if (it.data.isEmpty()) {
+                            binding.messageView.setup(
+                                R.drawable.elephant_friend_empty,
+                                R.string.message_empty,
+                                null
+                            )
+                            binding.messageView.show()
+                        } else {
+                            binding.messageView.hide()
+                            binding.recyclerView.show()
+                        }
+                    }
+                    is LoadState.Error -> {
+                        binding.progressBar.hide()
+                        binding.swipeRefreshLayout.isRefreshing = false
+                        binding.recyclerView.hide()
+                        if (adapter.itemCount != 0) {
+                            val snackbar = Snackbar.make(
+                                binding.root,
+                                it.throwable.message ?: "Error",
+                                Snackbar.LENGTH_INDEFINITE
+                            )
+
+                            if (it.throwable !is HttpException || it.throwable.code() != 404) {
+                                snackbar.setAction("Retry") { viewModel.accept(InfallibleUiAction.Reload) }
+                            }
+                            snackbar.show()
+                        } else {
+                            if (it.throwable !is HttpException || it.throwable.code() != 404) {
+                                binding.messageView.setup(it.throwable) {
+                                    viewModel.accept(InfallibleUiAction.Reload)
+                                }
+                            } else {
+                                binding.messageView.setup(it.throwable)
+                            }
+                            binding.messageView.show()
+                        }
+                    }
+                }
+            }
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.statusDisplayOptions.collectLatest {
+                adapter.statusDisplayOptions = it
+            }
+        }
+
+        (activity as? ActionButtonActivity)?.actionButton?.hide()
```

Comment:
Can it every be an ActionButtonActivity here?

Reply:
Yes. `MainActivity`, which can host this as a tab, is an `ActionButtonActivity`.

Reply:
(can be resolved)